lib/OptimizeMiniRust.ml

```diff
@@ -1093,6 +1118,16 @@ let infer_mut_borrows files =
     We now apply the valuation to all functions to compute mutability inference. *)
   let files = List.map (fun (filename, decls) -> filename, List.map (function
     | Function _ as decl ->
+        (* We need to perform the inference twice to handle function parameters
+           that are becoming mutable borrows, and that are assigned to in the function.
+
+           The first execution will update the function type to mark the parameter as
+           a mutable variable of type mutable borrow.
+           The second execution will use this information when building the initial
+           `known` set, forcing expressions assigned to the mutable parameter to
+           be mutable borrows.
+        *)
+        let decl = infer_function env valuation decl in
```

Comment:
based on your comment it looks like what you want is a first List.map followed by a second one

Reply:
Correct, this is now fixed